apps/web/pages/signup.tsx

```diff
@@ -118,21 +140,31 @@ export default function Signup({ prepopulateFormValues, token, orgSlug }: Signup
                   if (methods.formState?.errors?.apiError) {
                     methods.clearErrors("apiError");
                   }
+
+                  if (methods.getValues().username === undefined && isOrgInviteByLink && orgAutoAcceptEmail) {
+                    methods.setValue(
+                      "username",
+                      getOrgUsernameFromEmail(methods.getValues().email, orgAutoAcceptEmail)
+                    );
+                  }
                   methods.handleSubmit(signUp)(event);
                 }}
                 className="bg-default space-y-6">
                 {errors.apiError && <Alert severity="error" message={errors.apiError?.message} />}
+                {}
                 <div className="space-y-4">
-                  <TextField
-                    addOnLeading={
-                      orgSlug
-                        ? `${getOrgFullDomain(orgSlug, { protocol: true })}/`
-                        : `${process.env.NEXT_PUBLIC_WEBSITE_URL}/`
-                    }
-                    {...register("username")}
-                    disabled={!!orgSlug}
-                    required
-                  />
+                  {!isOrgInviteByLink && (
+                    <TextField
+                      addOnLeading={
+                        orgSlug
+                          ? `${getOrgFullDomain(orgSlug, { protocol: true })}/`
+                          : `${process.env.NEXT_PUBLIC_WEBSITE_URL}/`
+                      }
+                      {...register("username")}
+                      disabled={!!orgSlug}
+                      required
+                    />
+                  )}
```

Comment:
hiding username field only in case of inviteByLink to org because we generate it from the email entered and it is disabled for user to change